### PR TITLE
HttpGetter with Basic Auth + tests

### LINKED ISSUE #32: HTTP Basic auth support?
Hi,

Any plans on adding parameters for http basic auth? I'm using Nomad and would like to download artifacts from a secured JFrog Artifactory.

thanks in advance!

### PR DESCRIPTION
**Pull request by adrianlop**
closes #32

### CODE REVIEW

**Comment by adrianlop** [comment]
hi @mitchellh can you review please?
I tried this code using Nomad master branch (and overwriting the go-getter in vendor).
It pulled a .jar from a secured JFrog Artifactory and worked.

**Comment by adrianlop** [comment]
btw, I can't reproduce the appveyor test errors (TestTarGzipDecompressor and TestZipDecompressor), all tests are passing in my dev env.
any hints please?

thanks!!

File: get_http.go [hunk 1]
```diff
@@ -115,6 +144,20 @@ func (g *HttpGetter) GetFile(dst string, u *url.URL) error {
 	return err
 }
 
+func (g *HttpGetter) parseUrl(u *url.URL) (basicAuthUser, basicAuthPass string) {
```
Comment (catsby) [inline get_http.go]: nit pick - this method name is pretty generic/broad but the actual implementation is pretty narrow, we're specifically grabbing the username/password, so a more descriptive name would be better

File: get_http.go [hunk 2]
```diff
@@ -115,6 +144,20 @@ func (g *HttpGetter) GetFile(dst string, u *url.URL) error {
...
+	qUser := "http_basic_auth_user"
+	qPass := "http_basic_auth_pass"
```
Comment (catsby) [inline get_http.go]: We need to document the format for users so that they can use this authentication.

File: get_http.go (hunk 1 above)
Comment (adrianlop) [inline get_http.go]: changed to `getHttpBasicAuth`

File: get_http.go (hunk 2 above)
Comment (adrianlop) [inline get_http.go]: added documentation in README.md

**Comment by adrianlop** [comment]
@catsby thanks for reviewing!
I didn't add support for `username:password@url.com`, it is discouraged and browsers like Chrome, Firefox, disabled support for this.

**Comment by adrianlop** [comment]
@catsby ping! can you review it again please? I'd like to do a PR to Nomad if this gets merged. thanks!

**Comment by catsby** [comment]
I think the changes look good to me, thanks! I defer to @mitchellh to 👍 / 👎 the merge

**Comment by holtwilkins** [comment]
Hey @adrianlop / @catsby / @mitchellh , happy new year!  Might it be possible to push this through so this functionality can make its way into the nomad artifact downloader?

**Comment by dadgar** [comment]
Closing in favor of #62

**Comment by dadgar** [comment]
@adrianlop @holtwilkins So go-getter already supports this with normal basic auth formatting:
`http://<user>:<password>@foo.com/bar`

**Pull request closed without merging**